homework/tests/example.spec.js

```diff
@@ -17,15 +17,115 @@ const data = [
+     });
+     test('Adding -3 and 0 results in -3', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('-3');
+      await page.locator('#number2Field').type('0');
+      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('-3');
+     });
+    });
+  test.describe(version + ': Subtract', () => {
+    test('Subtracting 5 and 4 results in 1', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('5');
+      await page.locator('#number2Field').type('4');
+      await page.selectOption('#selectOperationDropdown', {label: 'Subtract'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('1');
+    });
+  test.describe(version + ': Multiply', () => {
+     test('Multiplying 2 and 4 results in 8', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('2');
+      await page.locator('#number2Field').type('4');
+      await page.selectOption('#selectOperationDropdown', {label: 'Multiply'});
+      await page.locator('#calculateButton').click();
+   
+       await expect(page.locator('#numberAnswerField')).toHaveValue('8');
     });
   });
-});
\ No newline at end of file
+    test.describe(version + ': Divide', () => {
+      test('Dividing 6 and 2 results in 3', async ({ page }) => {
+       await page.goto('https://testsheepnz.github.io/BasicCalculator');
+       await page.selectOption('#selectBuild', { label: version});
+       await page.locator('#number1Field').type('6');
+       await page.locator('#number2Field').type('2');
+       await page.selectOption('#selectOperationDropdown', {label: 'Divide'});
+       await page.locator('#calculateButton').click();
+   
+       await expect(page.locator('#numberAnswerField')).toHaveValue('3');
+      });
+    });
+    test.describe(version + ': Concatenate', () => {
+      test('Concatenating 1 and 2 results in 12', async ({ page }) => {
+       await page.goto('https://testsheepnz.github.io/BasicCalculator');
+       await page.selectOption('#selectBuild', { label: version});
+       await page.locator('#number1Field').type('1');
+       await page.locator('#number2Field').type('2');
+       await page.selectOption('#selectOperationDropdown', {label: 'Concatenate'});
+       await page.locator('#calculateButton').click();
+   
+       await expect(page.locator('#numberAnswerField')).toHaveValue('12');
+      });
+    });
+
```

Comment:
Let's take this last test as an example for requirement: "**When test results in failure, actual result should be understandable from the test error message**.".

This timeout message "Test timeout of 10000ms exceeded." does not really provide a lot of information for us. When the test fail like we should check what is an issue, register it and improve the test so next time we will know that "Calculate" button is missing just looking into the report.

This could be achieved by using if() & else() blocks, for example:

`      if (await page.locator('#number1Field').isVisible())
      {
        await page.locator('#number1Field').type('2');
      }
      else{
        throw "First number input field not working/missing";
      }  
`
Now if you would use that, message "First number input field not working/missing" would be displayed in the report, instead of timeout message. However, this is not a good practice to use those checks in every single test, so please update your code:
1. Try to exclude those checks into the separate function and call it in every test, if those checks fails - further test code will not be executed
2. If you will not be able to come up with solution described in 1st point, update only your last test "Concatenating 1 and 2 results in 12" with those checks.

![image](https://user-images.githubusercontent.com/73665598/180774719-29a1faa3-e626-4816-9fcc-23fa1dcf08d8.png)